src/hooks/useRuntimeHealth.ts

```diff
@@ -0,0 +1,98 @@
+import { useQuery } from "@livestore/react";
+import { queryDb } from "@livestore/livestore";
+import { RuntimeSessionData, tables } from "@runt/schema";
+import { useCallback } from "react";
+
+export type RuntimeHealth =
+  | "healthy"
+  | "warning"
+  | "connecting"
+  | "disconnected"
+  | "unknown";
+
+export interface RuntimeHealthState {
+  activeRuntime: RuntimeSessionData | undefined;
+  hasActiveRuntime: boolean;
+  runtimeHealth: RuntimeHealth;
+  runtimeStatus: string;
+  runningExecutions: any[];
+  executionQueue: any[];
+}
+
+export const useRuntimeHealth = (): RuntimeHealthState => {
+  const runtimeSessions = useQuery(
+    queryDb(tables.runtimeSessions.select().where({ isActive: true }))
+  );
+
+  // Get execution queue for runtime health monitoring
+  const executionQueue = useQuery(
+    queryDb(tables.executionQueue.select().orderBy("id", "desc"))
+  ) as any[];
+
+  // Get running executions with SQL filtering for better performance
+  const runningExecutions = useQuery(
+    queryDb(
+      tables.executionQueue
+        .select()
+        .where({
+          status: { op: "IN", value: ["executing", "pending", "assigned"] },
+        })
+        .orderBy("id", "desc")
+    )
+  ) as any[];
+
+  // Check runtime status
+  const getRuntimeHealth = useCallback(
+    (session: RuntimeSessionData): RuntimeHealth => {
+      if (session.status === "starting") {
+        // If session is starting, it's connecting
+        return session.isActive ? "connecting" : "unknown";
+      }
+      if (!session.isActive) {
+        return "disconnected";
+      }
+      // For active sessions, use status to determine health
+      switch (session.status) {
+        case "ready":
+        case "busy":
+          return "healthy";
+        case "restarting":
+          return "warning";
+        case "terminated":
+          return "disconnected";
+        default:
+          return "unknown";
+      }
+    },
+    []
+  );
+
+  const activeRuntime = runtimeSessions.find(
+    (session: RuntimeSessionData) =>
+      session.status === "ready" || session.status === "busy"
+  );
```

Comment:
Looking at all this now makes me realize we could be optimizing for more SQL oriented operations at query time (on LiveStore) instead of computing it here in JS.